### PR TITLE
[#641] Add EditPlanNameModal for editing name and description of completed plans

### LINKED ISSUE #641: Edit Name and Description of Completed and Archived Migration Plans
Migration Plans in the Completed and Archived states should have an Edit option in their kebab menu which opens a simple modal for editing their Name and Description only.

### PR DESCRIPTION
**Pull request by mturley**
Closes #641.
Depends on https://github.com/ManageIQ/manageiq/pull/17989.

fwqwyj1f9e

### CODE REVIEW

**Comment by AparnaKarve** [comment]
@mturley would it be possible to display the "Plan Name already exists" alert at top of the modal?

**Comment by mturley** [comment]
@AparnaKarve I could, but I didn't want to have to add a bunch of whitespace above the name field or have the contents move around vertically when the alert is shown.  In the Plan Wizard we have the wizard steps as a natural margin to lay this alert over, but displaying it at the top as-is would cover up the name field.

@vconzola, I'm curious about your thoughts on this. Should we leave the alert at the bottom of the modal, or add sufficient margin above the name field so it can appear at the top?

**Comment by vconzola** [comment]
@mturley @AparnaKarve I'm not sure we need the inline message at all. I think the text below the input text box is probably sufficient if you added "plan name already exists" after "Enter a unique name" And drop the "Please" at the beginning. We're not supposed to use "please" in our UIs.

**Comment by vconzola** [comment]
@mturley OK, here's the deal. Inline notification is only needed for server side validation of form information. In our case, we're validating as information is entered into the field so a field error message only is appropriate. So you can do what I said above: "Plan name already exists. Enter a unique name."

**Comment by mturley** [comment]
This one should now be all set. The "Please" was already in place for the two wizards, and I was reusing the validation helper from the plan wizard where it was defined, so I went ahead and updated the text for the mapping wizard too. ("Plan name already exists. Enter a unique name." and "Mapping name already exists. Enter a unique name.")

screenshot 2018-10-04 14 28 44

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardGeneralStep/helpers.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ export const asyncValidate = (values, dispatch, props) =>
     const duplicateIsEditingPlanName = editingPlan && duplicateName === editingPlan.name;
 
     if (duplicateName && !duplicateIsEditingPlanName) {
-      props.showAlertAction(sprintf(__('Name %s already exists'), newPlanName));
-      const error = { name: 'Please enter a unique name' };
+      if (props.showAlertAction) props.showAlertAction(sprintf(__('Name %s already exists'), newPlanName));
+      const error = { name: 'Plan name already exists. Enter a unique name.' };
```
Comment (AparnaKarve) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardGeneralStep/helpers.js]: `__('Plan name already exists. Enter a unique name.')`

**Comment by AparnaKarve** [comment]
@mturley Can you make that one little gettext change above?
Everything else looks good!

**Comment by mturley** [comment]
Whoops, thanks @AparnaKarve! Done.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit d7ade0ec3ed0f7c5ca0a8af5e74bdf9897a2ac97
Author: Aparna Karve <aparna.karve@gmail.com>
Date:   Thu Oct 4 14:13:23 2018 -0700

    Merge pull request #663 from mturley/641-edit-plan-name
    
    [#641] Add EditPlanNameModal for editing name and description of completed plans
    
    (cherry picked from commit 424c024e395dd76e0d891b7a6694e184bde3367c)
```

**Pull request merged**